internal/planner/planner.go

```diff
@@ -2286,10 +2286,28 @@ outer:
 			break
 		}
 
-		for _, node := range nodes {
-			if len(node.Rules()) > 0 {
-				p.debugf("ref %s: at least one rule has 1+ rules, break", ref[0:index+1])
-				break outer
+		// NOTE(sr): we only need this check for the penultimate part:
+		// When planning the ref data.pkg.a[input.x][input.y],
+		// We want to capture this situation:
+		//   a.b[c] := "x" if c := "c"
+		//   a.b.d := "y"
+		//
+		// Not this:
+		//   a.b[c] := "x" if c := "c"
+		//   a.d := "y"
+		// since the length doesn't add up. Even if input.x was "d", the second
+		// rule (a.d) wouldn't contribute anything to the result, since we cannot
+		// "dot it".
```

Comment:
We're leaning a bit onto the compiler's restrictions wrt overlapping rules here... it's also a bit of a mess 😬